libs/remix-tests/src/testRunner.ts

```diff
@@ -84,6 +84,30 @@ function getAvailableFunctions (fileAST: AstNode, testContractName: string): str
     return funcList;
 }
 
+function getAssertMethodLocation (fileAST: AstNode, testContractName: string, functionName: string, assertMethod: string): string {
+    let location
+    if(fileAST.nodes && fileAST.nodes.length > 0) {
+        const contractAST: AstNode = fileAST.nodes.find(node => node.name === testContractName && node.nodeType === 'ContractDefinition')
+        if(contractAST && contractAST.nodes) {
+            const funcNode: AstNode = contractAST.nodes.find(node => (node.name === functionName && node.nodeType === "FunctionDefinition"))
+            // Check if statement nodeType is 'ExpressionStatement', for example:
+            // Assert.equal(foo.get(), 100, "initial value is not correct");
+            // Check if statement nodeType is 'Return', for example:
+            // return Assert.equal(foo.get(), 100, "initial value is not correct");
+            const expressions = funcNode.body.statements.filter(s => 
```

Comment:
This is super difficult to read. It looks to me that you're doing a lot of filtering operations. If I were you I would create a bunch of reusable guard functions with nice naming: 
```typescript
function isStatement(node: AstNode): node is AstStatement {
  return node.nodeType === 'ExpressionStatement';
}
```
The example above might not make sense in your case. It's just to showcase the concept.

Reply:
Not sure if type guards are suitable here. Added helpers.